<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>epitopefindr  • epitopefindr</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>


<!-- docsearch -->
<script src="docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>



<meta property="og:title" content="epitopefindr " />
<meta property="og:image" content="https://brandonsie.github.io/epitopefindr/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-131446137-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-131446137-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">epitopefindr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.30</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="articles/epitopefindr_avarda_vignette.html">epitopefindr Application: Viral Peptides</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Links
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://brandonsie.github.io/epitopefindr/news/index.html">Changelog</a>
    </li>
    <li>
      <a href="https://github.com/brandonsie/epitopefindr/">Github Repo</a>
    </li>
    <li>
      <a href="https://brandonsie.github.io/">brandonsie.github.io</a>
    </li>
  </ul>
</li>
      </ul>
      
      <form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>epitopefindr </h1>
    </div>


<div id="epitopefindr-" class="section level1">

<p><a href="https://www.gnu.org/licenses/gpl-3.0.en.html"><img src="https://img.shields.io/badge/licence-GPL--3-blue.svg" alt="License"></a> <a href="https://www.tidyverse.org/lifecycle/"><img src="https://img.shields.io/badge/lifecycle-maturing-blue.svg" alt="Lifecycle"></a> <img src="https://img.shields.io/github/languages/code-size/brandonsie/epitopefindr.svg" alt="Code Size"><img src="https://img.shields.io/github/last-commit/brandonsie/epitopefinder.svg" alt="Last Commit"></p>
<p>The purpose of this package is to describe the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> alignments among a set of peptide sequences by reporting the overlaps of each peptide’s alignments to other peptides in the set. One can imagine inputting a list of peptides enriched by immunoprecipitation (e.g. by <a href="https://www.nature.com/articles/s41596-018-0025-6">PhIP-seq</a>) to identify corresponding epitopes.</p>
<p><code>epitopefindr</code> takes a .fasta file listing peptide sequences of interest and calls BLASTp from within R to identify alignments among these peptides. Each peptide’s alignments to other peptides are then simplified to the minimal number of “non overlapping” intervals* of the index peptide that represent all alignments to other peptides reported by BLAST. (*By default, each interval must be at least 7 amino acids long, and two intervals are considered NOT overlapping if they share 6 or fewer amino acids). After the minimal overlaps are identified for each peptide, these overlaps are gathered into aligning groups based on the initial BLAST. For each group, a multiple sequence alignment logo (motif) is generated to represent the collective sequence. Additionally, a spreadsheet is written to list the final trimmed amino acid sequences and some metadata.</p>
<p><img src="https://raw.githubusercontent.com/brandonsie/brandonsie.github.io/master/resources/EpitopeFindRWorkflow2c.png" alt="workflow"></p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation:</h1>
<ol>
<li>Install <a href="https://www.r-project.org/">R (version 3.5+)</a>.<br>
</li>
<li>Install <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">BLAST+ (version 2.7.1)</a>. (Note: we have observed some issues with more recent versions of BLAST+ and will monitor for bugfixes.)</li>
<li>In R console, execute:</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co">if</span> (<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"devtools"</span>)) <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
<span class="kw">devtools</span>::<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"brandonsie/epitopefindr"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">epitopefindr</span>)
</pre></div>
<div id="optional-suggested-additional-setup-" class="section level2">
<h2 class="hasAnchor">
<a href="#optional-suggested-additional-setup-" class="anchor"></a>Optional (Suggested) Additional Setup :</h2>
<p><em>(These are not essential to <code>epitopefindr</code>, but are used to generate alignment logo PDFs from the alignment data, which can be valuable visualizations.)</em></p>
<ol>
<li>Install a TeX distribution with <code>pdflatex</code>. (e.g. <a href="https://miktex.org">MiKTeX</a>, <a href="https://tug.org/texlive/">Tex Live</a>). <em>(Optional; used to convert multiple sequence alignment TeX files to PDF.)</em><br>
</li>
<li>Install <a href="https://www.pdflabs.com/tools/pdftk-server/">pdftk (version 2.02+)</a>. <em>(Optional; used to merge individual PDFs into a single file.)</em> If you are unable to install pdftk, but your system has the pdfunite command line utility, you can install the R package <a href="https://github.com/brandonsie/pdfuniter"><code>pdfuniter</code></a>, which performs a similar function. With <code>pdfuniter</code>, run <code>epfind</code> with <code>pdftk = FALSE, pdfunite = TRUE</code>.</li>
</ol>
</div>
<div id="debugging" class="section level2">
<h2 class="hasAnchor">
<a href="#debugging" class="anchor"></a>Debugging</h2>
<ul>
<li>epitopefindr 1.1.29 (2020-05-30) updates the DESCRIPTION file to specify sources of Bioconductor and Github packages. If the above installation produces issues during certain package installations, try the following:</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co">if</span> (<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>)) <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
<span class="kw">BiocManager</span>::<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Biostrings"</span>, <span class="st">"IRanges"</span>, <span class="st">"msa"</span>, <span class="st">"S4Vectors"</span>))

<span class="co"># Install Github packages</span>
<span class="co">if</span>(<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"devtools"</span>)) <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
<span class="kw">devtools</span>::<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"mhahsler/rBLAST"</span>) 
<span class="kw">devtools</span>::<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"brandonsie/pdfuniter"</span>)   
<span class="kw">devtools</span>::<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"brandonsie/epitopefindr"</span>)
</pre></div>
<hr>
</div>
</div>
<div id="guide" class="section level1">
<h1 class="hasAnchor">
<a href="#guide" class="anchor"></a>Guide</h1>
<ol>
<li>Prepare a list of your peptides of interest using one of the following two methods. Either of these can be fed as the first input parameter to <code>epfind</code>.
<ul>
<li>Make a <a href="https://zhanglab.ccmb.med.umich.edu/FASTA/">FASTA file</a> with peptide names and sequences.</li>
<li>Make an <code>AAStringSet</code> object of peptides (identifier + sequence) as described in the <a href="http://web.mit.edu/~r/current/arch/i386_linux26/lib/R/library/Biostrings/html/XStringSet-class.html">Biostrings documentation</a>.</li>
</ul>
</li>
<li>To run a typical <code>epitopefindr</code> pipeline, try calling <code>epfind</code>:</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># Basic call</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">epfind</span>(<span class="op">&lt;</span>path to .fasta<span class="op">&gt;</span>, <span class="op">&lt;</span>path to output dir<span class="op">&gt;</span>)</span>
<span id="cb3-3"><a href="#cb3-3"></a></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co"># Without pdflatex or pdftk</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="kw">epfind</span>(<span class="op">&lt;</span>path to .fasta<span class="op">&gt;</span>, <span class="op">&lt;</span>path to output dir<span class="op">&gt;</span>, </span>
<span id="cb3-6"><a href="#cb3-6"></a>        <span class="dt">pdflatex =</span> <span class="ot">FALSE</span>, <span class="dt">pdftk =</span> <span class="ot">FALSE</span>)</span>
<span id="cb3-7"><a href="#cb3-7"></a></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co"># More stringent e-value threshold</span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="kw">epfind</span>(<span class="op">&lt;</span>path to .fasta<span class="op">&gt;</span>, <span class="op">&lt;</span>path to output dir<span class="op">&gt;</span>, <span class="dt">e.thresh =</span> <span class="fl">0.0001</span>)</span></code></pre></div>
<p>You can try running <code><a href="reference/epfind.html">epfind()</a></code> with some provided example data:</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">my_peptides</span> <span class="op">&lt;-</span> <span class="kw">epitopefindr</span>::<span class="kw"><a href="reference/pairwise_viral_hits.html">pairwise_viral_hits</a></span>
<span class="kw">epitopefindr</span>::<span class="fu"><a href="reference/epfind.html">epfind</a></span>(data = <span class="kw">my_peptides</span>, output.dir = <span class="st">"my_epf_1/"</span>)
</pre></div>
<p>A brief summary of the functions called by <code>epfind</code>:</p>
<ul>
<li>
<code>pbCycleBLAST</code> cycles through each input peptide to find the overlap of its alignment with other peptides from the input. Nested within a call to <code>pbCycleBLAST</code> are calls to <code>epitopeBLAST</code>, <code>indexEpitopes</code>.</li>
<li>
<code>trimEpitopes</code> performs a second pass through the identified sequences to tidy alignments.</li>
<li>
<code>indexGroups</code> collects trimmed sequences into aligning groups</li>
<li>
<code>groupMSA</code> creates a multiple sequence alignment motif logo for each group</li>
<li>
<code>outputTable</code> creates a spreadsheet summarizing identified sequences and epitope groups</li>
</ul>
<div id="for-more-information-please-visit" class="section level2">
<h2 class="hasAnchor">
<a href="#for-more-information-please-visit" class="anchor"></a>For more information, please visit:</h2>
<ul>
<li>
<a href="https://brandonsie.github.io/epitopefindr/">epitopefindr’s pkgdown website</a><br>
</li>
<li><a href="https://brandonsie.github.io/epitopefindr/articles/epitopefindr_avarda_vignette.html">vignette: epitopefindr Application: Viral Peptides</a></li>
</ul>
</div>
</div>


  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>



      <footer>
      <div class="copyright">
  <p>Developed by Brandon Sie.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: 'c0f1e745bc83170b812c2f3ddbeb17c6',
    indexName: 'epitopefindr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


