<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>All-in-one function to call the major steps of epitopefindr. — epfind • epitopefindr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous"><link href="../docsearch.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="All-in-one function to call the major steps of epitopefindr. — epfind"><meta property="og:description" content="All-in-one function to call the major steps of epitopefindr."><meta property="og:image" content="https://brandonsie.github.io/epitopefindr/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-131446137-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-131446137-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epitopefindr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.31</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/epitopefindr_avarda_vignette.html">epitopefindr Application: Viral Peptides</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Links
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="https://brandonsie.github.io/epitopefindr/news/index.html">Changelog</a>
    </li>
    <li>
      <a href="https://github.com/brandonsie/epitopefindr/" class="external-link">Github Repo</a>
    </li>
    <li>
      <a href="https://brandonsie.github.io/" class="external-link">brandonsie.github.io</a>
    </li>
  </ul></li>
      </ul><form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off"></div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>All-in-one function to call the major steps of epitopefindr.</h1>
    
    <div class="hidden name"><code>epfind.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>All-in-one function to call the major steps of epitopefindr.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">epfind</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  output.dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  e.thresh <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  g.method <span class="op">=</span> <span class="st">"any"</span>,</span>
<span>  aln.size <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  min.groupsize <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  min.consensus.pos <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  consensus.type <span class="op">=</span> <span class="st">"Biostrings"</span>,</span>
<span>  consensus.thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">49</span><span class="op">)</span>,</span>
<span>  peptide.nchar <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  msa.width <span class="op">=</span> <span class="st">"dynamic"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pdflatex <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pdftk <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  pdfuniter <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  make.png <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  name.msa <span class="op">=</span> <span class="st">"msa.pdf"</span>,</span>
<span>  name.alignments <span class="op">=</span> <span class="st">"finalAlignments.csv"</span>,</span>
<span>  name.epitopekey <span class="op">=</span> <span class="st">"epitopeKey.csv"</span>,</span>
<span>  name.epitopesum <span class="op">=</span> <span class="st">"epitopeSummary.csv"</span>,</span>
<span>  use.doParallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  delete.intermediates <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">epFind2</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  output.dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  e.thresh <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  g.method <span class="op">=</span> <span class="st">"any"</span>,</span>
<span>  aln.size <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  min.groupsize <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  min.consensus.pos <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  consensus.type <span class="op">=</span> <span class="st">"Biostrings"</span>,</span>
<span>  consensus.thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">49</span><span class="op">)</span>,</span>
<span>  peptide.nchar <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  msa.width <span class="op">=</span> <span class="st">"dynamic"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pdflatex <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pdftk <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  pdfuniter <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  make.png <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  name.msa <span class="op">=</span> <span class="st">"msa.pdf"</span>,</span>
<span>  name.alignments <span class="op">=</span> <span class="st">"finalAlignments.csv"</span>,</span>
<span>  name.epitopekey <span class="op">=</span> <span class="st">"epitopeKey.csv"</span>,</span>
<span>  name.epitopesum <span class="op">=</span> <span class="st">"epitopeSummary.csv"</span>,</span>
<span>  use.doParallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  delete.intermediates <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data</dt>
<dd><p>Biostrings::AAStringset input sequences to search for epitopes, or path to corresponding .fasta file.</p></dd>


<dt>output.dir</dt>
<dd><p>Directory to which output files should be written.</p></dd>


<dt>e.thresh</dt>
<dd><p>Maximum e-value to consider from BLASTp alignments of 'data'.</p></dd>


<dt>g.method</dt>
<dd><p>Grouping method of alignments. Either 'any' or 'all'. See ?indexGroups</p></dd>


<dt>aln.size</dt>
<dd><p>Minimum length of alignment to consider from BLASTp alignments of 'data'.</p></dd>


<dt>min.groupsize</dt>
<dd><p>Minimum number of peptides per group to require in order to print a group.</p></dd>


<dt>min.consensus.pos</dt>
<dd><p>Minimum number of amino acid consensus positions required in order to print a group.</p></dd>


<dt>consensus.type</dt>
<dd><p>"upperlower" or "Biostrings" type of msa consensus sequence to output</p></dd>


<dt>consensus.thresh</dt>
<dd><p>Two decreasing numeric values of upper and lower thresholds for sequence consensus.</p></dd>


<dt>peptide.nchar</dt>
<dd><p>Maximum of character from peptide name to use in msa output. Default 50. Starts from left.</p></dd>


<dt>msa.width</dt>
<dd><p>Controls whether or not MSA images have fixed or dynamic width. By default, msa.width is set to "dynamic", which causes the document dimentions of the resultant image to be calculated based on the lentht of the peptide name and the number of amino acids in the sequence alignment. If msa.width is instead set to a numeric, then an MSA will be printed with a fixed with that number of inches. With 50-character peptide.nchar and a maximum expected sequence alignment of 45 positions, an msa.width of 12 is more than sufficient.</p></dd>


<dt>verbose</dt>
<dd><p>Logical to print progress updates.</p></dd>


<dt>pdflatex</dt>
<dd><p>Logical whether or not to produce PDF LaTeX figures using pdflatex</p></dd>


<dt>pdftk</dt>
<dd><p>Logical whether or not to merge msa pdfs using staplr and pdftk</p></dd>


<dt>pdfuniter</dt>
<dd><p>Logical whether or not to merge msa pdfs using staplr and pdfuniter</p></dd>


<dt>make.png</dt>
<dd><p>Locial whether or not to convert PDF output to PNG.</p></dd>


<dt>name.msa</dt>
<dd><p>Filename for output merged pdf of msa logos.</p></dd>


<dt>name.alignments</dt>
<dd><p>Filename for output spreadsheet of peptide alignments.</p></dd>


<dt>name.epitopekey</dt>
<dd><p>Filename for output spreadhseet of epitopes per peptide.</p></dd>


<dt>name.epitopesum</dt>
<dd><p>Filename for output summary sheet of epitopes.</p></dd>


<dt>use.doParallel</dt>
<dd><p>Logical whether or not to use doParallel parallelization.</p></dd>


<dt>delete.intermediates</dt>
<dd><p>Logical whether or not to delete the intermediate_files folder at the end</p></dd>

</dl></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brandon Sie.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'c0f1e745bc83170b812c2f3ddbeb17c6',
    indexName: 'epitopefindr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script></body></html>

